habitat/sims/pyrobot/__init__.py

```diff
@@ -0,0 +1,17 @@
+def _try_register_pyrobot():
+    try:
+        import pyrobot
+
+        has_pyrobot = True
+    except ImportError as e:
+        has_pyrobot = False
+        pyrobot_import_error = e
+
+    if has_pyrobot:
+        from habitat.sims.pyrobot.pyrobot import PyRobot
+    else:
+
+        @registry.register_simulator(name="PyRobot-v0")
+        class PyRobotImportError(Simulator):
+            def __init__(self, *args, **kwargs):
+                raise pyrobot_import_error
```

Comment:
Shouldn't this happen inside the PyRobot class itself? I don't believe we need a separate check.

Reply:
This patten lets you use `pyrobot` types in the type annotations in the definition of the class and in other related things.

Reply:
If pyrobot is present in your PYTHONPATH, type annotator will automatically pick it up. Type annotations can be an issue for the custom created classes like `PyRobot` here. The import for `PyRobot` class should still happen in `__init__.py` but point of failure for initialization without PyRobot should be in init. Also, to use this class with type annotation you would still need to export it with `__all__` property.

Reply:
Yeah.  The point of failure become when you try to initialize the simulator.  The reason to have a class dedicated to the failure is that we want habitat to work correctly even if you don't have pyrobot installed.  We can't use the one we instantiate normally as if that has any pyrobot types on annotations, those will fail!